csrc/paged_attention/attention/attention_generic.cuh

```diff
@@ -0,0 +1,64 @@
+/*
+ * Adapted from https://github.com/NVIDIA/FasterTransformer/blob/release/v5.3_tag/src/fastertransformer/kernels/decoder_masked_multihead_attention_utils.h
```

Comment:
Have we changed anything in any of these files or is everything under paged_attention copy/paste from vllm?
If it's all copy/paste, can we use some kind of git submodule instead?
If not, can we highlight where exactly we got it from and at what commit hash? Maybe we could add a top level README in the paged_attention directory?

If we have changed anything, we should highlight what has changed, and possibly add unit tests for those changes. If we expect to change anything in the future, we may need to import any existing tests of this code so that we can ensure we don't break it.

Reply:
This was copy and pasted from vLLM, however we have removed a lot of it that we are not using (I'll find the exact version this was from) I think we keep it here as is, and put a comment for what version of vllm it is. One reason for this is I anticipate we will be making some changes in the future.

Reply:
This was from version 0.2.7, I added a comment in the kernels.

Reply:
if I control-f I'm not seeing "0.2.7" anywhere. maybe a short README in the `paged_attention` directory would help explain where it came from / how to maintain?